### ISSUE TITLE
Enhancement: [strict-boolean-expressions] allowAlwaysTrueConditions

### ISSUE THREAD

**Issue by peterHakio** [typescript-eslint/typescript-eslint#11318]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/strict-boolean-expressions/

### Description

By using "strict-boolean-expressions" you are implicitly to enabling "no-unnecessary-condition" since:
```ts
const a = {}
if (a) {
  console.log("a")
}
```
will raise an error for "Unexpected object value in conditional. The condition is always true                                  @typescript-eslint/strict-boolean-expressions"

I would like to have an option to not get these warnings. I practice I have a legacy code base with 1000 errors of this type if I enable it. So i would very much like to be able to enable or disable this feature. So that I can you the feature for what is meant for. 

### Fail

```TypeScript
const a = {}
const b:number|null = 0
if (a) {
  console.log(b?"a":"b")
}
```

### Pass

```TypeScript
const replace = 'me';
const a = {}
if (a) {
  console.log("a")
}
```

**Comment by kirkwaiblinger** [comment]
I'm curious if you would benefit from eslint's new [bulk suppressions feature](https://eslint.org/docs/latest/use/suppressions) more than from the proposed option. Have you tried that out? 

I think it's hard to see the current rule behavior not being desirable in any code you're newly writing, but it totally makes sense that it will flag lots of stuff you can't safely change in an existing codebase.

**Comment by bradzacher** [comment]
I would agree - I don't particularly see why you'd want this in a new codebase, but for migrating an old codebase it would be useful.

But we also do not build options to aid migrations because options live forever and are often forgotten to be turned off once the migration is done. Additionally options change how _new_ code is written - which is undesirable.

Suppressions are a much better way to do migrations as they allow you to start at the stricter behaviour whilst allowing the violations. Over time you can easily push to remove the suppressions by fixing errors.

**Comment by peterHakio** [comment]
We all agree that is not desirable for new code. But if this feature was available it would be a no brainer to enable, because most other things is auto fixable. From a design perspective it seems very strange that in order to use "strict-boolean-expression" you have to enable another seperate setting "no-unnecessary-condition". Especially is  "no-unnecessary-condition" is not fixable, while "strict-boolean-expression" for the most part is. 

This is not even mentioned in the documentation either other than a refence to "strict-boolean-expression" which is its own setting.

**Comment by bradzacher** [comment]
> We all agree that is not desirable for new code

To be clear -- it's not desirable for either new OR old code.

> From a design perspective it seems very strange that in order to use "strict-boolean-expression" you have to enable another seperate setting "no-unnecessary-condition".

Why is that strange?
The two rules are _entirely_ independent.
Rules are designed to be used isolated from one another and are intended to have zero knowledge of one another -- that's a core design principle of ESLint.

Additionally:
`strict-boolean-expressions` is an optional rule -- we do not include it in our shared configs -- you have to turn it on in your codebase manually.
`no-unnecessary-condition` is an optional rule and we enable it in just the `strict-type-checked` config.

**Comment by kirkwaiblinger** [comment]
Fun fact, there actually aren't any auto fixes in the current version of strict-boolean-expressions anymore, see https://github.com/typescript-eslint/typescript-eslint/pull/10548
